### PR TITLE
[PLATFORM-315] Copy snippet dialog

### PR DESCRIPTION
**Pull request by juhah**
Implemented copy snippet dialog for streams. Includes shared `CodeSnippet` component adapted from #226. Can be tested in the streams listing page.

screen shot 2019-02-07 at 12 50 43
screen shot 2019-02-07 at 12 50 53

### CODE REVIEW

File: app/src/shared/components/CodeSnippet/index.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    className,
+    children,
+}: Props) => (
+    <SyntaxHighlighter
+        language={language}
+        style={githubGist}
+        wrapLines={wrapLines}
+        showLineNumbers={showLineNumbers}
+        className={cx(styles.codeSnippet, className)}
+    >{children}{/* eslint-disable-line react/jsx-closing-tag-location */}</SyntaxHighlighter>
```
Comment (tumppi) [inline app/src/shared/components/CodeSnippet/index.jsx]: Hmm, why is this `eslint-disable` needed 🤔

File: app/src/userpages/components/SnippetDialog/index.jsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            copy(snippets[selectedLanguage])
+
+            this.setState({
+                copied: true,
+            })
+            if (this.timeout) {
+                clearTimeout(this.timeout)
+            }
+            this.timeout = setTimeout(() => {
+                this.setState({
```
Comment (tumppi) [inline app/src/userpages/components/SnippetDialog/index.jsx]: Maybe we should implement this `copy` functionality in a self-contained component? This pattern of controlling the `copy`/`copied` text appears to be occurring in at least a couple of different places.

File: app/src/shared/components/CodeSnippet/index.jsx (hunk 1 above)
Comment (juhah) [reply app/src/shared/components/CodeSnippet/index.jsx]: Putting the tags on a new line gave also extra lines in the snippet so this is to get around the error that the tag should be placed on the next line.

File: app/src/userpages/components/SnippetDialog/index.jsx (hunk 2 above)
Comment (juhah) [reply app/src/userpages/components/SnippetDialog/index.jsx]: Yes, good idea! Were you thinking something like this?

```
  <CopyProvider>
  	{(copyFn, copied) => (
  		<button onClick={copyFn}>{copied ? 'Copy' : 'Copied'}</button>
  	)}
  </CopyProvider>
```

Comment (tumppi) [reply app/src/userpages/components/SnippetDialog/index.jsx]: That would work 👌The provider should also take care of handling the timeout.

Comment (juhah) [reply app/src/userpages/components/SnippetDialog/index.jsx]: Might actually be more work than it initially seems :) I'll leave it out for now.

File: app/src/userpages/components/SnippetDialog/snippetDialog.pcss [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+.footer {
+  display: flex;
+  width: 100%;
+}
+
+.language {
+  span {
+    color: #0324FF;
```
Comment (fonty1) [inline app/src/userpages/components/SnippetDialog/snippetDialog.pcss]: This should be font-size: 14px

File: app/src/userpages/components/SnippetDialog/snippetDialog.pcss [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  }
+}
+
+.library {
+  margin-left: 1rem;
```
Comment (fonty1) [inline app/src/userpages/components/SnippetDialog/snippetDialog.pcss]: Hit the gas on this, maybe go up to `1.5rem` 😎

**Comment by fonty1** [comment]
Switching to Javascript -> Java -> Javascript shakes up the horizontal line for some reason: 

screenshot 2019-02-11 at 12 34 09

**Comment by timoxley** [comment]
@juhah I noticed the tabs style changed in this PR:
https://www.chromaticqa.com/snapshot?appId=5bf8ea538165920024d322ff&id=5c656a49601dd4002412c686

Is this correct?

**Pull request merged**